### PR TITLE
Read testgrid config from protobuf

### PR DESCRIPTION
**Pull request by hoegaarden**
As https://github.com/kubernetes/release/issues/709 the location and format of testgrid's config changed.
Currently, `gcbmgr` & `find_green_build` do not work, not even with `--build-at-head` -- currently only running with a specific commit seems to work, e.g. `--buildversion=v1.14.2-beta.0.85+66049e3b21efe1`

To be able to use the new format, this is an initial WIP.

We now need to figure out, how we go about that -- specifically the following points are important:
- everytime [config.proto](https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.proto) in [test-infra](git.k8s.io/test-infra) we need to make sure that we pick that up
- comparing the old mechanism to find the blocking test to the new one shows, that the new one considers way more tests. Is this all correct or do we somehow need to filter them

Also, to make testing the changes a bit easier, it is now possible to run the release tools from a different repo/branch:
```
# RELEASE_TOOL_BRANCH=proto-foo RELEASE_TOOL_REPO=https://github.com/pivotal-k8s/release ./gcbmgr stage release-1.14 --official --build-at-head
```

More to come ...

/cc @kubernetes/patch-release-team 
/cc @idealhack @Bubblemelon 

/hold

### CODE REVIEW

**Comment by idealhack** [comment]
Finally, we are writing new release tools in Go 🎉

File: gcb/release.yaml [hunk 1]
```diff
@@ -7,12 +7,17 @@ secrets:
...
-  - "https://github.com/kubernetes/release"
+  - "--branch=${_RELEASE_TOOL_BRANCH}"
+  - "${_RELEASE_TOOL_REPO}"
 
 - name: gcr.io/$PROJECT_ID/k8s-cloud-builder
-  dir: release
+  dir: "go/src/k8s.io/release"
+  env:
+  - "RELEASE_TOOL_REPO=${_RELEASE_TOOL_REPO}"
+  - "RELEASE_TOOL_BRANCH=${_RELEASE_TOOL_BRANCH}"
```
Comment (idealhack) [inline gcb/release.yaml]: are these still necessary in the future?

File: cmd/blocking-testgrid-tests/main.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	binName  = "testgrid-conf-reader"
+	usageFmt = `usage: %[1]s [release]
+	e.g. %[1]s release-1.14
+`
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Fprintf(os.Stderr, usageFmt, binName)
+		return
```
Comment (idealhack) [inline cmd/blocking-testgrid-tests/main.go]: `os.Exit(1)`

File: gcb/release.yaml (hunk 1 above)
Comment (hoegaarden) [reply gcb/release.yaml]: TL;DR: We need this (at least for now) for every `anago` call

We need this every time we call `anago` and want the to support alternative repos/branches/refs ... And I don't see why we wouldn't want this to be configurable, esp. now that we need to touch the release tooling and some major refactoring is about to happen soon.

They are necessary right now, as `anago` pretty much on every call does check if the repo is in the state it is supposed to be. Besides the fact that this check does not make too much sense on GCB (the first step on GCB is to checkout that repo from that branch and to that ref anyway), I didn't want to touch that right now and keep the current behaviour. On the long run we will probably remove that check anyway.

Comment (idealhack) [reply gcb/release.yaml]: Thanks for the answer! I agree that we need this option for refactoring, let's keep it.

I understand the git ref check is for the part run locally, so not sure about if we want to remove it in the future.

**Comment by hoegaarden** [comment]
Thanks @idealhack , PTAL

I reshuffled the history a bit. Some things I didn't change (e.g. I have the dependency imports still as its own commit for easier review), but mostly, I think, I addressed your overall concerns / suggestions.

**Comment by idealhack** [comment]
@hoegaarden big +1 on keep the dep commits alone, could you squash https://github.com/kubernetes/release/pull/715/commits/e1db13913c37c541c654cc1af73f87e0992ebec1 to previous ones too? otherwise LGTM

**Comment by hoegaarden** [comment]
PTAL @idealhack

**Pull request merged**